rs/nervous_system/candid_utils/src/printing/tests.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidStruct {
+    pub status: Option<i32>,
+    pub module_hash: Vec<u8>,
+    pub controllers: String,
+    pub memory_size: Option<u64>,
+    pub cycles: Option<u64>,
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+enum DummyCandidVariant {
+    Foo(String),
+    Bar { abc: String, xyz: String },
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidContainer {
+    foo: DummyCandidVariant,
+    bar: Result<DummyCandidVariant, String>,
+}
+
+#[track_caller]
+fn run_test<T: CandidType>(value: T, expected_result: Result<String, String>) {
```

Comment:
re name: How about "assert_pretty_string_eq"?

Even though this is defined within the same module, a less generic name would be good (from the call site, it is not clear where this is defined).